### PR TITLE
added task cleaning

### PR DESCRIPTION
**Pull request by KarlSchiller**
- cleaning has to be done from monday to saturday
- cleaning needs four workers each day
- the penalty for allocating cleaning to a worker twice has to be very high

### CODE REVIEW

File: soak-shiftplanning.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
 parser.add_argument('teilnehmer')
 parser.add_argument('--seed', type=int, default=0)
 
-num_workers = 3
```
Comment (maxnoe) [inline soak-shiftplanning.py:11]: Maybe

```
num_workers = defaultdict(lambda: 3)
num_workers['cleaning'] = 4
```

And looking up num_workers by task would be more elegant.

File: soak-shiftplanning.py [hunk 2]
```diff
@@ -130,9 +133,10 @@ def create_week_plan(workers, semesters):
     for day in days:
         weekplan.append({})
 
-        for meal in meals[day]:
-            assigned_workers_meal = pick_workers(shifts, workers, meal, weekplan)
-            weekplan[-1][meal] = list(assigned_workers_meal)
+        for task in tasks[day]:
+            num_workers = num_workers_cleaning if task=='cleaning' else num_workers_meals
```
Comment (maxnoe) [inline soak-shiftplanning.py]: see comment above

File: soak-shiftplanning.py [hunk 3]
```diff
@@ -155,7 +159,7 @@ def main():
     print('Starting iteration')
 
     counter = 0
-    max_allowed_score = 255 * len(workers)
+    max_allowed_score = 530 * len(workers)  # 255 without cleaning task
```
Comment (maxnoe) [inline soak-shiftplanning.py]: maybe make this a constant at the top?

File: soak-shiftplanning.py [hunk 4]
```diff
@@ -169,24 +173,23 @@ def main():
             continue
 
         total_score = sum(w['total_score'] for w in shifts.values())
-        print(total_score)
-        print(total_score / len(workers))
+        print(f"\ttotal score {total_score}")
```
Comment (maxnoe) [inline soak-shiftplanning.py:177]: nice

**Review by maxnoe — CHANGES_REQUESTED** [review]
Looks good!

Maybe change the two num_workers_* to a defaultdict?

File: soak-shiftplanning.py (hunk 1 above)
Comment (KarlSchiller) [reply soak-shiftplanning.py:11]: Covered by following commit

File: soak-shiftplanning.py (hunk 3 above)
Comment (KarlSchiller) [reply soak-shiftplanning.py]: Done in following commit

**Review by maxnoe — APPROVED** [review]

**Pull request merged**